jena-arq/src/main/java/org/apache/jena/riot/other/G.java

```diff
@@ -0,0 +1,593 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.jena.riot.other;
+
+import java.util.*;
+import java.util.function.Consumer;
+
+import org.apache.jena.atlas.iterator.Iter;
+import org.apache.jena.graph.Graph;
+import org.apache.jena.graph.GraphUtil;
+import org.apache.jena.graph.Node;
+import org.apache.jena.graph.Triple;
+import org.apache.jena.riot.out.NodeFmtLib;
+//import org.apache.jena.shacl.lib.GN;
+//import org.apache.jena.shacl.lib.RDFDataException;
+//import org.apache.jena.shacl.lib.Transitive;
+//import org.apache.jena.shacl.sys.C;
+import org.apache.jena.sparql.core.Quad;
+import org.apache.jena.sparql.graph.NodeConst;
+import org.apache.jena.sparql.util.graph.GNode;
+import org.apache.jena.sparql.util.graph.GraphList;
+import org.apache.jena.util.iterator.ExtendedIterator;
+
+/** A library of functions for working with {@link Graph}. */
+public class G {
+    private G() {}
+    private static Node rdfType = NodeConst.nodeRDFType;
+
+    /** Return the subject of a triple, or null if the triple is null. */
+    public static Node subject(Triple triple) {
+        return triple == null ? null : triple.getSubject();
+    }
+
+    /** Return the predicate of a triple, or null if the triple is null. */
+    public static Node predicate(Triple triple) {
+        return triple == null ? null : triple.getPredicate();
+    }
+
+    /** Return the object of a triple, or null if the triple is null. */
+    public static Node object(Triple triple) {
+        return triple == null ? null : triple.getObject();
+    }
+
+    // ---- Node filter tests.
+    public static boolean isURI(Node n)         { return n != null && n.isURI(); }
+    public static boolean isBlank(Node n)       { return n != null && n.isBlank(); }
+    public static boolean isLiteral(Node n)     { return n != null && n.isLiteral(); }
+    public static boolean isResource(Node n)    { return n != null && (n.isURI()||n.isBlank()); }
+
```

Comment:
How about isNodeTriple?

Reply:
Good point. You can see that the code is dated!
(and `isNodeGraph` even if it isn't current used)